src/main/java/org/opensearch/security/dlic/rest/api/TenancyConfigAction.java

```diff
@@ -0,0 +1,104 @@
+import org.opensearch.security.auditlog.AuditLog;
+import org.opensearch.security.configuration.AdminDNs;
+import org.opensearch.security.configuration.ConfigurationRepository;
+import org.opensearch.security.dlic.rest.validation.AbstractConfigurationValidator;
+import org.opensearch.security.dlic.rest.validation.TenancyConfigValidator;
+import org.opensearch.security.dlic.rest.validation.SecurityConfigValidator;
+import org.opensearch.security.privileges.PrivilegesEvaluator;
+import org.opensearch.security.securityconf.impl.CType;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+import org.opensearch.security.ssl.transport.PrincipalExtractor;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.threadpool.ThreadPool;
+
+import static org.opensearch.security.dlic.rest.support.Utils.addRoutesPrefix;
+
+public class TenancyConfigAction extends PatchableResourceApiAction{
+
+    private static final List<Route> allRoutes = new ImmutableList.Builder<Route>()
+            .addAll(addRoutesPrefix(
+                    ImmutableList.of(
+                            new Route(Method.GET, "/tenancyconfig/"),
+                            new Route(Method.PUT, "/tenancyconfig/"),
+                            new Route(Method.PATCH, "/tenancyconfig/")
+                    )
+            ))
+            .build();
+
+    @Inject
+    public TenancyConfigAction(Settings settings, Path configPath, RestController controller, Client client,
+                               AdminDNs adminDNs, ConfigurationRepository cl, ClusterService cs,
+                               PrincipalExtractor principalExtractor, PrivilegesEvaluator evaluator,
+                               ThreadPool threadPool, AuditLog auditLog) {
+        super(settings, configPath, controller, client, adminDNs, cl, cs, principalExtractor, evaluator, threadPool, auditLog);
+    }
+
+    @Override
+    public List<Route> routes() {
+        return allRoutes;
+    }
+
+    @Override
+    protected void handleGet(RestChannel channel, RestRequest request, Client client, final JsonNode content) throws IOException{
+
+        final SecurityDynamicConfiguration<?> configuration = load(getConfigName(), true);
+        filter(configuration);
+        successResponse(channel,configuration);
+    }
+
+
+    @Override
+    protected AbstractConfigurationValidator getValidator(RestRequest request, BytesReference ref, Object... params) {
+        return new TenancyConfigValidator(request, ref, this.settings, params);
+    }
+
+    @Override
+    protected void handleDelete(RestChannel channel, final RestRequest request, final Client client, final JsonNode content) throws IOException{
+        notImplemented(channel, Method.DELETE);
+    }
+
+    @Override
+    protected void handlePut(RestChannel channel, final RestRequest request, final Client client, final JsonNode content) throws IOException{
+        successResponse(channel);
```

Comment:
If this is not doing anything, shouldn't it be `notImplemented` as well ?

Reply:
We want to enable PUT request for tenancyConfig. That's why we are not using `notImplemented ` .

Reply:
For my understanding, why is it returning successResponse directly?